### DISCUSSION TITLE
Setting up Allowed Domains that can interact with my API

### DISCUSSION THREAD

**Discussion by ayodyln** [nitrojs/nitro#2278] [Q&A]
Hello 😄

I'm using Nitro to create a web service for my company. It's a fairly simple service that takes in form submissions and handles them. **However, I want to only allow a set amount of domains that could use my API.** How can I go about configuring this properly?

_I have read through Nitro and H3 documentation to find a solution and I nothing is working from what I can tell._

For example,

I only want a certain website that has a form only be allowed to submit a POST request, and reject all others.

**Comment by JCtapuk** [comment]
> Hello 😄
> 
> I'm using Nitro to create a web service for my company. It's a fairly simple service that takes in form submissions and handles them. **However, I want to only allow a set amount of domains that could use my API.** How can I go about configuring this properly?
> 
> _I have read through Nitro and H3 documentation to find a solution and I nothing is working from what I can tell._
> 
> For example,
> 
> I only want a certain website that has a form only be allowed to submit a POST request, and reject all others.

Doesn't the Origin and Host headers help?

**Reply by ayodyln** [reply]
> > Hello 😄
> 
> > 
> 
> > I'm using Nitro to create a web service for my company. It's a fairly simple service that takes in form submissions and handles them. **However, I want to only allow a set amount of domains that could use my API.** How can I go about configuring this properly?
> 
> > 
> 
> > _I have read through Nitro and H3 documentation to find a solution and I nothing is working from what I can tell._
> 
> > 
> 
> > For example,
> 
> > 
> 
> > I only want a certain website that has a form only be allowed to submit a POST request, and reject all others.
> 
> 
> 
> Doesn't the Origin and Host headers help?

I'm still learning the best practices here. I know I can set the headers in my event handler. But I tested it and it wasn't doing any errors for a domain that isn't on my allowed domain list. 

I'll look those up and research more. 

Are headers supposed to be set for both the response and request? Do the request headers dictate if a response will be sent?
